plugins/c8y_firmware_plugin/src/firmware_manager.rs

```diff
@@ -534,11 +521,45 @@ impl FirmwareManager {
         );
 
         if op_state == ActiveOperationState::Pending {
-            self.mqtt_client.published.send(executing_msg).await?;
+            self.mqtt_client
+                .published
+                .send(executing_msg)
+                .await
+                .map_err(MqttError::from)?;
         }
 
-        self.mqtt_client.published.send(failed_msg).await?;
+        self.mqtt_client
+            .published
+            .send(failed_msg)
+            .await
+            .map_err(MqttError::from)?;
+
+        Ok(())
+    }
+
+    // This function can be removed once we start using operation ID from c8y.
+    fn request_is_already_addressed(
```

Comment:
```suggestion
    fn is_same_request_in_progress(
```
As we're just validating if the same request is already being processed using just the live operation files and not historical data. Even if the same request was processed earlier and completed, we won't be able to figure that out as we don't keep historical data.